src/middlewared/middlewared/restful.py

```diff
@@ -497,9 +511,31 @@ def get_post_args(self, req, resp, **kwargs):
                     filterid = kwargs['id']
                     if filterid.isdigit():
                         filterid = int(filterid)
+                    extra = {}
+                    text = await req.text()
```

Comment:
Are we reading request body of a GET method?

Reply:
```
 if not method_args:
                # RFC 7231 specifies that a GET request can accept a payload body
                # This means that all the http methods now ( delete, get, post, put ) accept a payload body
```
Yes

Reply:
That does not make it a good request semantic. You can accept any JSON variable just fine using query parameters `/items/query-options.extra.include_chart_schema=true&query-options.extra.retrieve_resources=false`